### PR TITLE
Bugfix/newlines

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

File: doxia-modules/doxia-module-markdown/src/test/resources/test.md [hunk 1]
```diff
@@ -2,3 +2,12 @@ Markdown test document
 ======================
 
 Placeholder for future expansion.
+
+- item1
+- item2
+- item3
+    - item31
```
Comment (michael-o) [inline doxia-modules/doxia-module-markdown/src/test/resources/test.md:9]: Isn't this too much indentation?

Comment (kwin) [reply doxia-modules/doxia-module-markdown/src/test/resources/test.md:9]: https://docs.newrelic.com/docs/style-guide/structure/styleguide-markup-indentation/#two-space

Comment (kwin) [reply doxia-modules/doxia-module-markdown/src/test/resources/test.md:9]: 4 spaces is most compatible with different markdown implementations and works under all circumstances.

**Review by slachiewicz — APPROVED** [review]

**Comment by slachiewicz** [comment]
Do You have plans to backport it to Doxia 1.x ?

**Comment by kwin** [comment]
I would rather update doxia-converter to 2.x.

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java [hunk 2]
```diff
@@ -144,21 +149,18 @@ protected String getFooterBlock() {
...
-                + MarkdownMarkup.LIST_ORDERED_ITEM_START_MARKUP + ""
-                + Markup.SPACE + getEscapedText(item) + EOL + EOL;
+        return MarkdownMarkup.LIST_ORDERED_ITEM_START_MARKUP + "" + Markup.SPACE + getEscapedText(item) + EOL + EOL;
     }
 
     /** {@inheritDoc} */
     protected String getDefinitionListBlock(String definum, String definition) {
-        return EOL + EOL + Markup.SPACE + "" + Markup.LEFT_SQUARE_BRACKET + definum + Markup.RIGHT_SQUARE_BRACKET + ""
-                + Markup.SPACE + definition + EOL + EOL;
+        return "<dl>" + EOL + "<dt>" + getEscapedText(definum) + "</dt>" + EOL + "<dd>" + getEscapedText(definition)
```
Comment (michael-o) [inline doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java:162]: Why are litaral tags here?

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java [hunk 3]
```diff
@@ -212,24 +214,23 @@ protected String getDivisionBlock(String text) {
...
     /** {@inheritDoc} */
     protected String getHorizontalRuleBlock() {
-        return EOL + MarkdownMarkup.HORIZONTAL_RULE_MARKUP + EOL;
+        return MarkdownMarkup.HORIZONTAL_RULE_MARKUP + EOL + EOL;
     }
 
     /** {@inheritDoc} */
     protected String getPageBreakBlock() {
-        return EOL + MarkdownMarkup.PAGE_BREAK_MARKUP + EOL;
+        return "";
```
Comment (michael-o) [inline doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java:233]: I would explicitly return `StringUtils#EMPTY`.

File: doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownMarkup.java [hunk 4]
```diff
@@ -92,10 +91,7 @@ public interface MarkdownMarkup extends TextMarkup {
     String VERBATIM_START_MARKUP = "```";
 
     /** Syntax for the non breaking space: "\ " */
-    String NON_BREAKING_SPACE_MARKUP = String.valueOf(BACKSLASH) + SPACE;
-
-    /** Syntax for the page break: "\f" */
-    String PAGE_BREAK_MARKUP = String.valueOf(PAGE_BREAK);
+    String NON_BREAKING_SPACE_MARKUP = "&npsp;";
```
Comment (elharo) [inline doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownMarkup.java:94]: should this be `&nbsp;`?

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java (hunk 3 above)
Comment (elharo) [reply doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java:233]: I prefer "". I definitely don't want to add additional dependencies on Plexus StringUtils

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java [hunk 5]
```diff
@@ -349,11 +350,34 @@ public void testMultipleAuthors() {
...
+    @Test
+    public void testRoundtrip() throws IOException, ParseException {
+        parseFile(parser, "test", getSink());
+
+        final SinkEventTestingSink regeneratedSink = new SinkEventTestingSink();
+        try (Reader reader = new StringReader(getSinkContent())) {
+            parser.parse(reader, regeneratedSink);
+        }
+
+        System.out.println(getSinkContent());
```
Comment (elharo) [inline doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java:367]: delete this line; passing tests should generate no output

File: doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/SinkEventElement.java [hunk 6]
```diff
@@ -80,4 +80,22 @@ public String toString() {
         builder.append(']');
         return builder.toString();
     }
+
+    @Override
+    public int hashCode() {
```
Comment (elharo) [inline doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/SinkEventElement.java:85]: These methods could use tests

**Review by elharo — CHANGES_REQUESTED** [review]

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java (hunk 2 above)
Comment (kwin) [reply doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java:162]: https://dev.to/scottshipp/don-t-use-non-test-constants-in-unit-tests-3ej0, I don't think declaring constants for the test only makes this clearer.

Comment (michael-o) [reply doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java:162]: You should then add a comment why there are literals for clarity.

File: doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/SinkEventElement.java (hunk 6 above)
Comment (kwin) [reply doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/SinkEventElement.java:85]: Those are test classes only! I don't think adding test for tester utilities is very useful. Tests will fail if those don't work!

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java (hunk 5 above)
Comment (kwin) [reply doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java]: Done in https://github.com/apache/maven-doxia/pull/146/commits/c49775264627b41daa3b17e90f9d72a0b0d9c218.

File: doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownMarkup.java (hunk 4 above)
Comment (kwin) [reply doxia-modules/doxia-module-markdown/src/main/java/org/apache/maven/doxia/module/markdown/MarkdownMarkup.java]: Fixed in https://github.com/apache/maven-doxia/pull/146/commits/c49775264627b41daa3b17e90f9d72a0b0d9c218.

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java (hunk 3 above)
Comment (kwin) [reply doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java:235]: I agree with @elharo. Using a constant for this seems like overkill.

File: doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java (hunk 2 above)
Comment (kwin) [reply doxia-modules/doxia-module-markdown/src/test/java/org/apache/maven/doxia/module/markdown/MarkdownSinkTest.java:164]: Done in https://github.com/apache/maven-doxia/pull/146/commits/c49775264627b41daa3b17e90f9d72a0b0d9c218.

**Review by elharo — APPROVED** [review]

**Review by michael-o — APPROVED** [review]

**Pull request merged**